pkgs/by-name/dc/dcgm/package.nix

```diff
@@ -54,44 +39,43 @@ let
     libcurand
   ];
 
-  # Builds CMake code to add CUDA paths for include and lib.
-  mkAppendCudaPaths = { version, pkgSet }:
+  # Builds CMake flags to add CUDA paths for include and lib.
+  mkCudaFlags = cudaPackages:
     let
+      version = cudaPackages.cudaMajorVersion;
       # The DCGM CMake assumes that the folder containing cuda.h contains all headers, so we must
       # combine everything together for headers to work.
-      # It would be more convenient to use symlinkJoin on *just* the include subdirectories
-      # of each package, but not all of them have an include directory and making that work
-      # is more effort than it's worth for this temporary, build-time package.
-      combined = symlinkJoin {
-        name = "cuda-combined-${version}";
-        paths = pkgSet;
+      headers = symlinkJoin {
+        name = "cuda-headers-combined-${version}";
+        paths = lib.map (pkg: "${lib.getInclude pkg}/include") (getCudaPackages cudaPackages);
       };
-      # The combined package above breaks the build for some reason so we just configure
-      # each package's library path.
-      libs = lib.concatMapStringsSep " " (x: ''"${x}/lib"'') pkgSet;
-    in ''
-      list(APPEND Cuda${version}_INCLUDE_PATHS "${combined}/include")
-      list(APPEND Cuda${version}_LIB_PATHS ${libs})
-    '';
+    in [
+      (lib.cmakeFeature "CUDA${version}_INCLUDE_DIR" "${headers}")
+      (lib.cmakeFeature "CUDA${version}_LIBS" "${cudaPackages.cuda_cudart.stubs}/lib/stubs/libcuda.so")
+      (lib.cmakeFeature "CUDA${version}_STATIC_LIBS" "${lib.getLib cudaPackages.cuda_cudart}/lib/libcudart.so")
+      (lib.cmakeFeature "CUDA${version}_STATIC_CUBLAS_LIBS" (lib.concatStringsSep ";" [
+        "${lib.getLib cudaPackages.libcublas}/lib/libcublas.so"
+        "${lib.getLib cudaPackages.libcublas}/lib/libcublasLt.so"
+      ]))
+    ];
 
 # gcc11 is required by DCGM's very particular build system
 # C.f. https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmbuild/build.sh#L22
```

Comment:
I guess `gcc11` is no longer strictly required?

Reply:
Ah, yes, I just forgot to drop this comment; it’s building with GCC 13 now. Fixed :)